registry/registry.go

```diff
@@ -200,6 +209,44 @@ func (registry *Registry) ListenAndServe() error {
 	return registry.server.Serve(ln)
 }
 
+// configureLetsEncryptChallenge adds a listener on port 80 for a "http-01" challenge.
+// This is required because Let's Encrypt disabled TLS-SNI-01 and TLS-SNI-02 challenges on 09.01.2018.
+// See: https://community.letsencrypt.org/t/important-what-you-need-to-know-about-tls-sni-validation-issues/50811
+func (registry *Registry) configureLetsEncryptChallenge(manager *autocert.Manager, port int) {
+	s := &http.Server{
+		Handler: manager.HTTPHandler(nil),
+	}
+
+	if port == 0 {
+		port = 80
+	}
+
+	s.Addr = fmt.Sprintf(":%d", port)
+
+	go func() {
+		dcontext.GetLogger(registry.app).Infof("listening on %v, letsencrypt", s.Addr)
+		dcontext.GetLogger(registry.app).Fatal(s.ListenAndServe())
+	}()
+}
+
+func resolveLetsEncryptCacheDirectory(cacheFile, cacheDir string) (string, error) {
+	if cacheFile != "" && cacheDir != "" {
+		return "", fmt.Errorf("cachefile and cachedir specified. choose one")
+	}
+
+	if cacheDir != "" {
+		return cacheDir, nil
+	}
+
+	if cacheFile != "" {
+		p := path.Dir(cacheFile)
```

Comment:
Is this backwards compatible?

Reply:
I tried to be as much as I can.

The problem is that `rsc.io/letsencrypt` stores certs in a single file as json, and `golang.org/x/crypto/acme/autocert` in one file per domain + extra file for acme related data :/

I know its not ideal, but how big of a problem is it as certs are going to auto-renew anyway?

My initial thought was to get rid of `cachefile` all together since letsencrypt is not working for new domains as it stands, but then I decided to be less intrusive than that :D

Reply:
Gotcha, so the previous cache file will not be used, but it will use the directory. What happens to the original cache file?

Reply:
The original cache file remains untouched. I didn't want to lump in clean-up tasks as well, but rather let the user decide what to when migrating to a newer version of the registry + LE.